### PR TITLE
Update subsetting for tibbles (#680)

### PR DESCRIPTION
**Pull request by mikemahoney218**
This is my attempt at rewriting the section around subsetting to consider the more consistent behavior of tibbles, which is what our novices will be principally using. 

There is a call-out that this behavior should only be expected when working with tibbles on line 289. I agree that it's not worth diving into the differences between base data frames and tibbles at this level, but I do want to at least flag "this is not universally true." If this feels likely to confuse, though, happy to remove it.

### CODE REVIEW

File: 02-starting-with-data.Rmd
Comment (Teebusch) [inline 02-starting-with-data.Rmd:269]: Very nice explanation!

File: 02-starting-with-data.Rmd
Comment (Teebusch) [inline 02-starting-with-data.Rmd:307]: ...or `tail(34779)`, I suppose  :)

**Review by Teebusch — APPROVED** [review]
Looks excellent!

(I think it's ok to talk about data frames rather than tibbles. We mention the difference briefly somewhere in the lesson nd say that it's not important.)

File: 02-starting-with-data.Rmd
Comment (mikemahoney218) [reply 02-starting-with-data.Rmd:307]: I tried my best to keep to a minimal rewrite :joy: 

If anything, it almost makes sense to move this one a little later in the lesson so that we can subset using `surveys[-7:nrow(surveys), ]` rather than hard-code the number of rows in the table, which feels a _little_ bit like teaching people to use "magic numbers" instead of semantic functions or variables. But this feels like a less clear case than the rest of the PR, so I just left it as it was :joy:

Comment (Teebusch) [reply 02-starting-with-data.Rmd:307]: Very good point about not using "magic numbers". I believe `nrow()` is mentioned earlier in this lesson (`inspecting data frames`), so it would be ok to use it here, wouldn't it?

I know you want to keep the rewrite minimal, but from my perspective using `nrow()` instead of `34786` is much better. And since you are already improving this section, you could just fix this along the way. 
 @fmichonneau , what do you think?

**Comment by mikemahoney218** [comment]
@Teebusch @fmichonneau I just pushed a commit with `nrow` replacing the "magic number" in the subsetting example, as we discussed last month -- not sure if it needs another review or not!

**Comment by mikemahoney218** [comment]
CI failed with `unable to load shared object '/home/runner/work/_temp/Library/stringi/libs/stringi.so'` which seems to be unrelated to these changes (all commits have failed in past few days).

**Review by Teebusch — APPROVED** [review]

File: 02-starting-with-data.Rmd
Comment (Teebusch) [reply 02-starting-with-data.Rmd:269]: I just noticed a typo: "to select ~the~ all columns for the first row:"

**Comment by Teebusch** [comment]
fixed a typo, minor changes to comments

**Review by Teebusch — APPROVED** [review]

**Comment by Teebusch** [comment]
I believe this PR can be merged. I'll leave it to one of the other maintainers to have a final look and push the button. :)

**Comment by fmichonneau** [comment]
This looks good! Thank you both for your work on this. I think introducing the double bracket might be a little beyond the scope of this lesson for the intended audience, but it could be useful in some contexts. Ideally, we'd want to add a mention of this in the instructor notes. Happy to merge as is though.

**Pull request merged**
